src/webots/nodes/utils/WbSupervisorUtilities.cpp

```diff
@@ -983,6 +1063,27 @@ void WbSupervisorUtilities::handleMessage(QDataStream &stream) {
             mImportedNodesNumber = importedNodesNumber;
           break;
         }
+        case WB_SF_NODE: {
+          QString filename = readString(stream);
+          makeFilenameAbsolute(filename);
+          int importedNodesNumber;
+          WbNodeOperations::OperationResult operationResult;
+          if (filename.endsWith(".wbo", Qt::CaseInsensitive))
+            operationResult =
+              WbNodeOperations::instance()->importNode(nodeId, fieldId, index, filename, "", &importedNodesNumber, true);
+          else {
+            operationResult = WbNodeOperations::FAILURE;
+            assert(false);
+          }
+          if (operationResult != WbNodeOperations::FAILURE)
+            mImportedNodesNumber = importedNodesNumber;
+          const WbSFNode *sfNode = dynamic_cast<WbSFNode *>(field->value());
+          assert(sfNode);
+          mImportedNodesNumber = -1;
```

Comment:
It seems useless to set `mImportedNodesNumber = importedNodesNumber;` at line 1079 and then overwrite the value here without using it in between.
Also note that it seems better to move `mImportedNodesNumber = -1;` in the else block of the `if (sfNode->value())` check to avoid a useless double variable set.